tiff_2_png/tiff_2_png_converter.py

```diff
@@ -235,33 +239,56 @@ def convert_to_png(
         ) as progress:
             for slice in progress:
                 _img = images_in_sequence[slice]
-                img_8_bit = np.subtract(_img, min_pixel)
-                img_8_bit = np.true_divide(_img, max_pixel - min_pixel)
-                images_in_sequence_8_bit[slice] = np.multiply(img_8_bit, 255).astype(
-                    np.uint8
-                )
+                img_8_bit = np.subtract(_img, global_min_pixel)
+                img_8_bit = np.true_divide(_img, global_max_pixel - global_min_pixel)
+                img_8_bit = np.multiply(img_8_bit, 255).astype(np.uint8)
+                if gamma is not None:
+                    img_8_bit = iph.gamma_correction(img_8_bit, gamma)
+                images_in_sequence_8_bit[slice] = img_8_bit
 
         image_out = build_image_sequence(
             images_in_sequence_8_bit, width, height, num_files_in_sequence, np.uint8
         )
 
-        name, extension = os.path.splitext(dest)
+        ## split path to resolve path of multiple files to be written
+        parent_folder = os.path.abspath(os.path.join(dest, os.pardir))
+        file_base_name = os.path.basename(dest)
+        file_name, extension = os.path.splitext(file_base_name)
         if not extension:
             extension = ".png"
 
         # export as single channel image
-        file_name = name + "_chn_" + str(channel) + extension
-        logging.info("##Saving channel " + str(channel) + " Image to: " + file_name)
-        cv2.imwrite(file_name, image_out)
-        file = open(name + "_metadata", "a")
+        file_name = file_name + "_chn_" + str(channel)
+        file_name_full_path = os.path.join(parent_folder, file_name + extension)
+        logging.info(
+            "##Saving channel " + str(channel) + " Image to: " + file_name_full_path
+        )
```

Comment:
Cleaner to use f string
```suggestion
        logging.info(
            f"##Saving channel {channel} image to: {file_name_full_path}"
        )
```

Reply:
I like this kind of format. It's better than adding strings.